### ISSUE TITLE
[Feature Request] (more detailed) Status page

### ISSUE THREAD

**Issue by henfri** [openmediavault/openmediavault#142]
Hello,

I made the experience, that it is very important to keep an eye on the server. Concretely:
1) when was the last successful/unsuccessful backup
2) when was the last successful/unsuccessful Snapraid Scrub
3) when was the last successful/unsuccessful Snapraid Sync
4) Any errors in the logs?

I am sure that there are many more.
Ideally, by selecting the particular entry, the log would be shown.

For sure, this requires the respective function (mostly: Plugin) to filter/report this information, but also a framework/place in the core of OMV to process this data.

The as-is situation for me is not satisfactory. I get many mails per day from cron jobs or plugins, usually showing me that everything is fine. This way, I stop reading the mails.

I think that the aim should be a 'lights out' philosophy:
If everything is fine, no notification is given. If something is wrong, this is reported (optionally) by mail and on the status page.

On point 4):
this is probably tricky. But I made good experience with logcheck (https://linux.die.net/man/8/logcheck)
It requires some tweaking though. By default, still lots of unneccessary messages are being not filtered out. 
But I think this can be achieved quite well with help by the community. The idea is to use logcheck to filter the syslog for the relevant messages (problems) only and display these in the Web-Interface.

Greetings,
Hendrik

**Comment by ryecoaaron** [comment]
Most of this would/should part of the particular plugin you are talking about.  What you probably really want is a monitoring system like zabbix.

**Comment by henfri** [comment]
Hello Aaron,

zabbix looks way to large.
Why do you think that zabbix is appropriate for the average OMV user?

I am not talking about a monitoring plugin, but about a facility in the core of OMV to show this information. The information must come from where the actual functionality (e.g. backup, snapraid, ...) is implemented, of course. This can be a plugin or part of core omv.

Don't you think, that this is useful or in fact needed?

How do you monitor the systems health?

Greetings,
Hendrik

**Comment by ryecoaaron** [comment]
Do I think zabbix is appropriate for the average OMV user?  No but your monitoring requests aren't typical of the average OMV user.  I was suggesting that for you.

OMV does have a dashboard with plugins that could be used to show the information you want.  I wrote one or two plugins for it but never went any further.

Myself, I interact with my systems often enough and I have trained myself to always read emails even if they are always good (I had no choice to do this for work).

I actually have enough systems at home that I should use zabbix (I use it at work too).

**Comment by henfri** [comment]
Hello,

I wonder what makes my requests untypical?
Shouldn't it be the intent of the average OMV user to see their systems status at a glance?

The dashboard currently has rather limited information. But it is the right place for this information.

What are the plugins you wrote for this? 
Don't you think that it should be the responsibility of the 'source' plugin to pre-process this information? Otherwise I think this would become a maintainance hell to keep the source and the monitor plugin in sync.

>Myself, I interact with my systems often enough and I have trained myself to always read emails even if they are always good (I had no choice to do this for work).

Interesting. At work, I cannot manage to read all my mails. Maybe that has become a habit.
But at work I also learned this light out philosophy (from aicraft cockpits. Just imagine that a 'light' could be 'normal' or a 'failure' with all these lights in an aircraft cockpit. That is why in an aircraft cockpit a light is always something asking for attention. This is what I am proposing here.

Greetings,
Hendrik

**Comment by votdev** [comment]
I would like to see Prometheus/Grafana Stack, but all Monitoring systems are oversized for OMV. Collectd and Monti must be enough, everything else does not match the target users and usage of OMV.

**Comment by henfri** [comment]
Hello Volker,

why do you think that there is no option inbetween the oversized monitoring systems and what OMV provides today?

My suggestions were quite concrete, I hoped. Let me rephrase:
1) Plugins should report their status to OMV. It can be the plugins task to analyse their status
2) Filter the syslog with logcheck and display the result in the OMV Userinterface
3) Information by Mail if errors appear in 1 or 2.

Don't you think that this would be very importand and still quite feasible?


Regards,
Hendrik

**Comment by ryecoaaron** [comment]
I say your request is "untypical" because many NAS users setup their NAS and forget about it until it stops working.  I would say a lot of OMV users don't check their dashboard or even login to the web interface monthly.

When I say plugin, I don't mean a full plugin.  Dashboard widget is the proper term.  I don't even remember what they were for but I can't find the code now.  Either way, it could be included with the plugin (if it was written).

Status lights are similar to the emails I get.  Most of my systems at work email on error only.  This is very possible to do with a lot of the plugins that email including snapraid, rsync, rsnapshot, etc.

While it might seem important for a few users, I personally don't think it is important enough for me to spend my little amount of time to implement.

**Comment by ryecoaaron** [comment]
By the way, I just spent an hour or two and setup a VM running zabbix (first time ever) to monitor two physical machines and three VMs.  Pretty easy to do.

**Comment by henfri** [comment]
Hello Aaron,

ok, I understand.
> forget about it until it stops working

This is what I want to improve by this proposal.

>Status lights are similar to the emails I get. Most of my systems at work email on error only. This is very possible to do with a lot of the plugins that email including snapraid, rsync, rsnapshot, etc.

I know. But it's not the default behaviour.
But it should be, don't you think?

What really needs to be avoided is daily 'here's the log. please check' mails. The machines should serve us, not the other way around.

> While it might seem important for a few users,

I think that few users would request it. But I thinkt that it could save a lot of users from trouble.

> , I personally don't think it is important enough for me to spend my little amount of time to implement.

This is something I can understand very well.
I could contribute on the logcheck side, if this is selected.

> By the way, I just spent an hour or two and setup a VM running zabbix (first time ever) to monitor two physical machines and three VMs. Pretty easy to do.

Did I trigger this ;-)
What is zabbix showing/reporting to you now? I would be surprised (but delighted) if it did points 1-4 from my first post. 

Greetings,
Hendrik

**Comment by votdev** [comment]
Your suggestions are ok, but most are related to Snapraid, so this plugin should implement that suggestions, this is not the job of the core system.

About the cron emails, it is not possible to tell cron when and when not to send emails. If there is command output and email is configured, then it is sent, no matter what it’s content.

**Comment by ryecoaaron** [comment]
rsnapshot has a checkbox to send on error only and that is the default option.  I guess I could write a short wrapper script to the snapraid sync command to only email on error but I never thought one daily snapraid email was that much of a nuisance.  It tells you the sync is running.

You reminded me that I wanted to install zabbix to have a better understanding for it at work.  I am just a zabbix user not an admin at work and wanted to know more.

My zabbix install is showing that one of my drives is almost full but I already knew that (it isn't a problem).  And zabbix could very easily do your 1-4.  It might require writing a short script or two but it definitely can.  We use it at work for MUCH more advanced monitoring than that.

**Comment by henfri** [comment]
Hello Volker,

thanks for your reply.
My list is not complete. Another example is mdadm. Another smart-values.
For smart the feature I have in mind is already implemented on the Dashboard in the way I have in mind. I thought that also problems would be reported by mail, but currently I cannot find this on the Webinterface.

> About the cron emails, it is not possible to tell cron when and when not to send emails.

A workaround can be to have a system mail account that is not forwarded to a real mail address but processed by the system before and then sent -only if something is to be reported.
This filtering of course can only be done by OMV if the cron-job is one that is defined by a plugin. OMV cannot know/process/interpret the output of the user-defined cronjobs.

> I never thought one daily snapraid email was that much of a nuisance. It tells you the sync is running.

Well, it makes me disregard it quickly. Especially as it is very long. I usually do not scroll down to the interesting bit anymore. 

And what's important is to get only one mail per day/week/whatever that summarizes the health of the system. Not one mail per subsubsystem.

Maybe zabbix would be my solution. But in fact I am also thinking about the (even) less advanced user and the 'completeness' of OMV.

Greetings,
Hendrik

**Comment by henfri** [comment]
Hello,

here is an example of the log-summary I would suggest for the more detailed status page:

> This email is sent by logcheck. If you no longer wish to receive
such mail, you can either uninstall the logcheck package or modify
its configuration file (/etc/logcheck/logcheck.conf).

System Events
=-=-=-=-=-=-=
Using fallback detection... please install python-magic for better gzip detection.
       0 Lines skipped (already processed)
       0 Patterns to ignore
       0 Ignored lines
       1 homeserver sshd: Accepted publickey for root from 192.168.177.9 port 57839 ssh2: RSA SHA256:z28fpZ5X8xhQIU9SpoKuaffVOLZwuwVUm53wKeVK5xA
       2 homeserver sshd: pam_unix(sshd:session): session opened for user root by (uid=0)
       1 homeserver sshd: Accepted publickey for root from 192.168.177.9 port 57840 ssh2: RSA SHA256:z28fpZ5X8xhQIU9SpoKuaffVOLZwuwVUm53wKeVK5xA
       1 homeserver smbd: [2018/07/22 20:36:14.492142,  0] ../source3/param/loadparm.c:3244(process_usershare_file)
       1 homeserver smbd:   process_usershare_file: stat of /var/lib/samba/usershares/hendrik failed. Keine Berechtigung
       1 homeserver su: + /dev/pts/2 root:logcheck

This is created by logcheck.
I would envision a summary of the last day/week/month to be selected.

Greetings,
Hendrik
